fvtest/tril/tril/ast.hpp

```diff
@@ -155,6 +213,21 @@ struct ASTValue {
      * b.isCompatibleWith<double>(); // returns true
      * b.isCompatibleWith<int>();    // returns false
      */
+#if __cplusplus <= 199711L
```

Comment:
So, there's two distinctions here: 

1. The use of `enable_if` rather than `std::enable_if` 
2. The use of `std::is_integral` directly, rather than via the alias `is_Integer_Compatible` 

@Leonardo2718 Why the indirection of is_Integer_Compatible? 

@rbanerjee If we were to to standardize on `std::is_integral`, this could collapse nicely with a `using std::enable_if<T>` (syntax may be wrong).

Reply:
The indirection is there mostly for two reasons:

1. In future, I was planning on making some changes to the type system in Tril, so having the indirection meant only having to change the code in one place.
2. Conceptually, I wanted there to be a distinction between the Tril type system and the C++ type system. The indirection was part of the interface for "bridging" the two, the goal being to hide as much of the details of how types are handled in the background behind a clear and precise interface.

Although I would prefer to keep the extra indirection for the sake of mental clarity, I'm okay with removing it as part of this change. In this case, I see no value in keeping both implementations, since it would defeat the goal of reducing code duplication.